scene/3d/skeleton_3d.cpp

```diff
@@ -329,32 +268,53 @@ void Skeleton3D::_notification(int p_what) {
 				}
 			}
 
-			dirty = false;
-
 #ifdef TOOLS_ENABLED
 			emit_signal(SceneStringNames::get_singleton()->pose_updated);
 #endif // TOOLS_ENABLED
 
 		} break;
 
+#ifndef _3D_DISABLED
 		case NOTIFICATION_INTERNAL_PHYSICS_PROCESS: {
 			// This is active only if the skeleton animates the physical bones
 			// and the state of the bone is not active.
-			if (animate_physical_bones) {
-				for (int i = 0; i < bones.size(); i += 1) {
-					if (bones[i].physical_bone) {
-						if (bones[i].physical_bone->is_simulating_physics() == false) {
-							bones[i].physical_bone->reset_to_rest_position();
+			if (Engine::get_singleton()->is_editor_hint()) {
+				if (animate_physical_bones) {
+					for (int i = 0; i < bones.size(); i += 1) {
+						if (bones[i].physical_bone) {
+							if (bones[i].physical_bone->is_simulating_physics() == false) {
+								bones[i].physical_bone->reset_to_rest_position();
+							}
 						}
 					}
 				}
 			}
+
+			if (modification_stack.is_valid()) {
+				execute_modifications(get_physics_process_delta_time(), SkeletonModificationStack3D::EXECUTION_MODE::execution_mode_physics_process);
+			}
+
 		} break;
+#endif // _3D_DISABLED
+
+#ifndef _3D_DISABLED
 		case NOTIFICATION_READY: {
```

Comment:
Not really important, but I would move this after `NOTIFICATION_INTERNAL_PROCESS`. In general we will do a check less, each frame, but however the code will look more consistent. Right now we have:
```c++
case NOTIFICATION_INTERNAL_PHYSICS_PROCESS
case NOTIFICATION_READY
case NOTIFICATION_INTERNAL_PROCESS
```

This should be better:
```c++
case NOTIFICATION_INTERNAL_PHYSICS_PROCESS
case NOTIFICATION_INTERNAL_PROCESS
case NOTIFICATION_READY
```

Reply:
Thanks!